platform/engineering/flag-for-manual-review.md

```diff
@@ -76,7 +94,11 @@ No other context is included.
 
 ### Security Concerns
 
-We should make sure that the Github bot auth token is kept secure and it should not be associated with an individual's account.  The sole communicators in the system are CircleCI and Github.  This isn't really a security concern, but more of a reliability one: if one of those services goes down we do not have a fallback plan.
+We should make sure that the Github bot auth token is kept secure and it should not be associated with an individual's account.
+The current strategy for storing this token is as an [environment variable in CircleCI](https://ui.circleci.com/settings/project/github/department-of-veterans-affairs/vets-website/environment-variables) called `GITHUB_TOKEN`.
```

Comment:
From the [CircleCI docs](https://circleci.com/docs/2.0/env-vars/):

> Note: secret masking will only prevent the value of the environment variable from appearing in your build output. The value of the environment variable is still accessible to users debugging builds with SSH.

who has SSH access to these builds?

Reply:
My understanding is that anyone who has an SSH public key on Github will be able to connect. I re-ran a job with SSH and this is what I see:

![image](https://user-images.githubusercontent.com/2008881/76005767-5621aa00-5ec0-11ea-9641-e65dfb465e9f.png)

So anytime we deliberately kick off a job with SSH access, we should be able to see whoever connects to it.

Reply:
_anyone_ with an SSH public key? So, that sounds to me like the GitHub bot auth token can be accessed by anyone?

Reply:
I found a more specific answer in [the docs](https://circleci.com/docs/2.0/ssh-access-jobs/):

> In order to SSH into a CircleCI build, the username must be one which has access to the project being built!

When I try to go to [the project website](https://app.circleci.com/pipelines/github/department-of-veterans-affairs/vets-website) in a private window it blocks me and forces me to authenticate with Github or Bitbucket.

I created a fake github account to test this and was able to login and view the project in Circle, but I can't even look at the project settings. I get errors when I try to follow the project, and I can't see the list of branches either.  The option to trigger an SSH build is disabled as well:

![image](https://user-images.githubusercontent.com/2008881/76033868-27242c00-5ef2-11ea-8d9e-52f26d3193d5.png)

After seeing all of this, I think that project access in circle requires an account to be a part of the Github organization that the repo is associated with. I've looked around at various settings to try to get a more definitive answer, but I'm not an organization admin in Circle so there are certain things I can't see. I wonder if [this page](https://circleci.com/gh/organizations/department-of-veterans-affairs/settings#projects) would be helpful.

Reply:
https://discuss.circleci.com/t/disable-ssh/31057 has a couple key things:

> Any user with write capabilities will also have SSH access. [...] any user with write permissions could theoretically also expose secrets via commands issues through the config.yml file.

Related question to investigate: how often do we hand out `write` permission on vets-website/vets-api, or do we generally keep people's access to `read`?

> A good first step is to ensure your deploy keys only have read access to your repos.

This made me wonder, what kinds of access does our bot token have? Can we keep that as minimal as possible? (read-only + posting comments?).

Other possible idea: it sounds like [contexts](https://circleci.com/docs/2.0/contexts/#restricting-a-context) might be useful here.

Reply:
Do you know who gives `write` access to people?

Our bot token has the [`public_repo` scope](https://developer.github.com/apps/building-oauth-apps/understanding-scopes-for-oauth-apps/#available-scopes).  I believe that this is the least permissions that we can give it in order for it to do its job, but it does include write access for our public repos.

> Limits access to public repositories. That includes read/write access to code, commit statuses, repository projects, collaborators, and deployment statuses for public repositories and organizations.

Here is a [relevant Slack thread](https://dsva.slack.com/archives/CJYRZK2HH/p1581623319018100?thread_ts=1581620875.013600&cid=CJYRZK2HH).

Contexts are something that I haven't explored deeply.  I'm interested to see how we could use them, but I don't have permissions to manage security groups so I'm a bit limited.

Reply:
> Do you know who gives `write` access to people?

VA has an entire GitHub admin group. you can also talk to @ricetj about longer-term strategy for ownership on these repos, or (more immediately helpful) see if you can track down an admin for `vets-website` and `vets-api` to see how we've handed out write access so far. it's possible that most people just have `read` access and that this isn't a concern, but `write` access is typically needed for managing issues, which might mean we've handed it out more broadly.

> Our bot token has the [`public_repo` scope](https://developer.github.com/apps/building-oauth-apps/understanding-scopes-for-oauth-apps/#available-scopes). I believe that this is the least permissions that we can give it in order for it to do its job, but it does include write access for our public repos.

hmm, this set of information is making me more and more concerned 😬 

what does the bot need from `public_repo` that it can't get from `(no scope)`?

Reply:
FWIW- Write access is needed by a GitHub team for codeowners to trigger so everyone has write access to the repo. 

All of the FE group on VSP has admin access. 

Generally everyone has `write` access.